test/runtests.jl

```diff
@@ -1,5 +1,64 @@
-using EcoBase
-using Base.Test
+using Compat
+using Compat.Test
+using Compat: @info
 
-# write your own tests here
-@test 1 == 2
+# Identify files in test/ that are testing matching files in src/
+#  - src/Source.jl will be matched by test/test_Source.jl
+
+filebase = map(file -> replace(file, r"(.*).jl$" => s"\1"),
+                filter(file -> occursin(r".*\.jl$", file),
+                       readdir("../src")))
+testbase = map(file -> replace(file, r"test_(.*).jl$" => s"\1"),
+                filter(str -> occursin(r"^test_.*\.jl$", str), readdir()))
+
+println()
+@info "Running tests for files:"
+for t in testbase
+    println("    = $t.jl")
+end
+println()
+
+@testset "* Testing $t.jl" for t in testbase
+    include("test_$t.jl")
+end
+
+# Identify tests with no matching file
+superfluous = filter(f -> f ∉ filebase, testbase)
+if length(superfluous) > 0
+    println()
+    @info "Potentially superfluous tests:"
+    for f in superfluous
+        println("    + $f.jl")
+    end
+    println()
+end
+
+# Identify files with no matching test
+missing = filter(f -> f ∉ testbase, filebase)
+if length(missing) > 0
+    println()
+    @info "Potentially missing tests:"
+    for f in missing
+        println("    - $f.jl")
+    end
+    println()
+end
+
+# Identify files that are cross-validating results against other packages
+# test/pkg_Package.jl should validate results against the Package package
+
+pkgbase = map(file -> replace(file, r"pkg_(.*).jl$" => s"\1"),
```

Comment:
What's this - looks interesting?

Reply:
It’s just a simple thing I have set up on my packages - any file starting test_ will be run first and is assumed to test the matching file in src/, and then any file beginning pkg_ is run and is assumed to do some validation against some external package. The latter may not be necessary here (or we could have a pkg_Diversity.jl and a pkg_SpatialEcology.jl)... any other files are ignored. It just helps me sort out in my head what I’m doing. By no means compulsory!